apps/www/_blog/2024-08-13-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,208 @@
+
+When you want to connect to a Realtime Channel, you do something close to the following code:
+
+```tsx
+// With authenticated user token
+client.realtime.setAuth(token)
+
+client.channel('channel').subscribe((status: string, err: any) => {
+  if (status === 'SUBSCRIBED') {
+    console.log('Connected')
+  }
+})
+```
+
+This will enable you to send messages but they will all be public. Any user would be subscribe to this Channel and read these messages if they had your anon token.
+
+## Adding Authorization to Realtime Channels
+
+To convert this into a Realtime Channel that checks RLS policies requires only two steps:
+
+- [Create RLS Policies](#create-rls-policies)
+- [Enabling Authorization on Channel subscription](#enabling-authorization-on-channel-subscription)
+
+### Create RLS policies
+
+Any RLS policy will work. Reference our [Github Discussion](github.com/orgs/supabase/discussions/22484) for more complex use cases.
+
+With this example we'll keep it simple and allow authenticated users to connect to a channel, send and receive Broadcast messages and track Presence.
+
+```sql
+create policy "authenticated user listen to all"
+on "realtime"."messages"
+as permissive for select
+to authenticated
+using ( true );
+
+create policy "authenticated user write to all"
+on "realtime"."messages"
+as permissive for insert
+to authenticated
+with check ( true );
+```
+
+We also have a new database function called `realtime.topic()` which enables you to write Policies that take into consideration the name of the Channel trying to run the RLS Policy.
+
+```sql
+create policy "authenticated users can only read from 'locked' topic"
+on "realtime"."messages"
+as permissive for select
+to authenticated
+using ( realtime.topic() = 'locked' );
+```
+
+### Enabling Authorization on Channel subscription
+
+Now that you have your RLS policies set, you will need to define on your Channel subscription that you intend to use a private channel. To achieve it we've introduced a new configuration parameter `private` to signal to Realtime servers that you want to check authorization on the channel.
+
+If you try to subscribe with an unauthorized user you will get a new error message informing the user that they do not have permission to access the topic.
+
+```tsx
+// With anon user
+supabase.realtime
+  .channel('locked', { config: { private: true } })
+  .subscribe((status: string, err: any) => {
+    if (status === 'CHANNEL_ERROR') {
+      console.error(err.message)
+    }
+  })
+
+// Outputs the following error:
+// "You do not have permissions to read from this Topic"
+```
+
+But if you connect with an authorized user you will be able to listen to all messages from the “locked” topic
+
+```tsx
+// With an authenticated user
+supabase.realtime.setAuth(token)
+
+supabase.realtime
+  .channel('locked', { config: { private: true } })
+  .subscribe((status: string, err: any) => {
+    if (status === 'CHANNEL_ERROR') {
+      console.error(err.message)
+    }
```

Comment:
@filipecabaco should change this since it's the same as the unauthorized case to status === 'CHANNEL_SUBSCRIBED' or whatever the successful status is console.log something like subscribed to 'locked' topic!